tests/mlir_converter_test.py

```diff
@@ -3,24 +3,12 @@
 docutils = pytest.importorskip("mlir")
 
 from xdsl.mlir_converter import MLIRConverter, mlir
-from xdsl.dialects.scf import Scf
-from xdsl.dialects.func import Func
-from xdsl.dialects.memref import MemRef
-from xdsl.dialects.affine import Affine
-from xdsl.dialects.arith import Arith
 from xdsl.parser import Parser
-from xdsl.dialects.builtin import Builtin, MLContext
+from xdsl.dialects.builtin import MLContext
 
 
 def convert_and_verify(test_prog: str):
     ctx = MLContext()
-    builtin = Builtin(ctx)
-    func = Func(ctx)
-    affine = Affine(ctx)
-    arith = Arith(ctx)
```

Comment:
I must admit, I am quite confused why this works. We are using operations from the arith dialect in the test, so it should be the case that we need to register the dialect in order to parse it. Maybe this goes through the unregistered operation parsing we recently added? pinging @math-fehr to got your perspective/idea.

Reply:
I have the feeling that this test is not actually passing, since it is not ran unless you have `mlir` python bindings imported.
The parser at least fail here with this input, we only allow unregistered operations when specified in the `Parser` constructor.

Reply:
The test is indeed not running

Reply:
To make the test run locally, you should use the [MLIR Python Bindings](https://mlir.llvm.org/docs/Bindings/Python/), with commit hash 74992f4a5bb79e2084abdef406ef2e5aa2024368 .
The reason why we do not run this on the CI, is that we would need to install MLIR to do so, which is really annoying and makes the CI quite slow.
In any way, this test won't pass without these dialects registered.

Reply:
right, ok, reverting